### PR TITLE
allow user to set hbm/ddr bandwidth in device topology

### PR DESCRIPTION
**Pull request by chengscott**

### CODE REVIEW

**Review by YLGH — APPROVED** [review]
changes look good to me, thanks

File: torchrec/distributed/planner/constants.py [hunk 1]
```diff
@@ -38,6 +38,8 @@
 def kernel_bw_lookup(
     compute_device: str,
     compute_kernel: str,
+    bw_hbm_mem: float,
+    bw_ddr_mem: float,
```
Comment (joshuadeng) [inline torchrec/distributed/planner/constants.py:42]: nit: rename to `hbm_mem_bw` and  `ddr_mem_bw`

File: torchrec/distributed/planner/shard_estimators.py [hunk 2]
```diff
@@ -146,6 +148,8 @@ def perf_func_emb_wall_time(
     input_data_type_size: float,
     output_data_type_size: float,
     num_poolings: List[float],
+    bw_hbm_mem: float,
+    bw_ddr_mem: float,
```
Comment (joshuadeng) [inline torchrec/distributed/planner/shard_estimators.py:152]: nit: ditto

File: torchrec/distributed/planner/shard_estimators.py [hunk 3]
```diff
@@ -121,6 +121,8 @@ def estimate(
                 input_data_type_size=BIGINT_DTYPE,
                 output_data_type_size=sharding_option.tensor.element_size(),
                 num_poolings=num_poolings,
+                bw_hbm_mem=self._topology.hbm_mem_bw,
+                bw_ddr_mem=self._topology.ddr_mem_bw,
```
Comment (joshuadeng) [inline torchrec/distributed/planner/shard_estimators.py:125]: nit: ditto

**Review by joshuadeng — COMMENTED** [review]
changes look good, thanks! left comment for a minor change

**Pull request closed without merging**